extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/AddNamespaceDecorator.java

```diff
@@ -43,4 +38,22 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(namespace);
     }
+
+    @Override
+    public void visit(KubernetesListBuilder list) {
+        List<HasMetadata> buildItems = list.buildItems()
+                .stream()
+                .filter(o -> o instanceof HasMetadata)
```

Comment:
Look at the code I sent previously, you should be able to simplify this even more and remove the `clusteredResources` set completely, if I'm not mistaken.